### ISSUE TITLE
"direct distribution" mode for experiments: same link for all participants

### ISSUE THREAD

**Issue by davidschlangen** [clp-research/slurk#84]
For some use cases (smaller experiments, piloting with a known set of users (e.g., participants in a class), it would be useful to be able to distribute an experiment (= task room type) to several users with the same URL.

When someone clicks on this URL, what should happen is that they get assigned a unique name, get put into the waiting room, and then, when the usual criteria are met (number of participants that are required), they get moved to a pre-determined task room.

In other words, the idea is that the URL does not encode a token that needs to be unique for each participant.

I'm fairly sure this is already possible (by putting a token into the URL that is re-usable? but does that result in new names for everyone who clicks on the link?), but it would be good if it could be documented prominently how to do it.

**Comment by TimDiekmann** [comment]
In fact, this is not possible at the moment. Token are reusable by default, until it is invalidated via `DELETE /api/v2/token/<string:id>`. However, a token is the current way to identify the user so that each client would be mapped to the same user.

I see two options here: 

- Use an external service, which maintain several tokens and redirekt the client to a new URL with a unique token. 
- Implement the above logic directly into slurk.

Could you assign a priority to this issue?

**Comment by davidschlangen** [comment]
I see pros and cons for both solutions. Wrapping an external service (a kind of broker) around an unchanged slurk has the advantage of, well, keeping slurk unchanged, but the disadvantage of making it necessary to start yet another component, which kind of defeats (some of) the purpose of making this an easy way to run non-crowdsourced experiments.

Am I understanding correctly that without a token, you can't even login? Otherwise a simple solution would be to just provide the link to the login page and let people assign a name to themselves?

**Comment by TimDiekmann** [comment]
> Am I understanding correctly that without a token, you can't even login?

Yes, without a token, no actions are possible at all.

**Comment by davidschlangen** [comment]
How substantial would be the changes if this were to be implemented on the server? (For example, as a special mode in which it is started, which would lead to every entry via a link creating a new user.)
